test/MauiTestUtils/src/DeviceTests/xUnitCustomizations.cs

```diff
@@ -0,0 +1,192 @@
+﻿#nullable enable
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace Microsoft.Maui
+{
+    /// <summary>
+    /// Custom trait discoverer which adds a Category trait for filtering, etc.
+    /// </summary>
+    public class CategoryDiscoverer : ITraitDiscoverer
+    {
+        public const string Category = "Category";
+
+        public IEnumerable<KeyValuePair<string, string>> GetTraits(IAttributeInfo traitAttribute)
+        {
+            var args = traitAttribute.GetConstructorArguments().FirstOrDefault();
+
+            if (args is string[] categories)
+            {
+                foreach (var category in categories)
+                {
+                    yield return new KeyValuePair<string, string>(Category, category.ToString());
```

Comment:
this is code copied from MAUI - I don't want to change that unless necessary to get it to work within our repo - so that it's easier to update later on.